api/geth_backend.go

```diff
@@ -1261,6 +1261,11 @@ func (b *GethStatusBackend) startWallet(watchNewBlocks bool) error {
 		}
 	}
 
+	err = wallet.MergeBlocksRanges(allAddresses, b.statusNode.Config().NetworkID)
```

Comment:
how expensive is this operation for large accounts?

Reply:
It depends on device I and the number of ranges, a bit hard to estimate... I thought about limiting it to 100-200 ranges per run, just to make sure we do not block the app for a few seconds in the worst case

Reply:
what about running it in the background, any implication?

Reply:
Right now it is run on starting wallet, and if there is no new transactions we will try to upsert a new scanned range quite soon after start. I guess the slowest part of that merging would be db transaction and it might lock table for write operations (at least I suppose it does) and thus insertion (or update) of a new range will be delayed. I'm not sure that's a big deal, also not sure whether that blocking will be noticeable in the app, but it looks like we will just move it from one place to another.

Reply:
ok, so we don't wait for `startWallet` to login etc , correct?
If we don't wait, yes, running in the background is not useful

Reply:
we don't wait and also soon we will run it only after user opens wallet tab so it might not happen at all :)